### PR TITLE
Fix data order of H0 in GRU Operator

### LINKED ISSUE #5421: Enable hidden state  initialization in GRU Operator.
There is a bug for the initialized hidden state in GRU operator now.

### PR DESCRIPTION
**Pull request by guoshengCS**
fix #5421

### CODE REVIEW

File: paddle/operators/gru_op.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #pragma once
 
+#include "paddle/operators/lstm_op.h"
```
Comment (qingqing01) [inline paddle/operators/gru_op.h]: Maybe write [`ReorderInitState`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/lstm_op.h#L28) again in this file, not include `lstm_op.h`.

File: paddle/operators/gru_op.h [hunk 2]
```diff
@@ -176,11 +194,13 @@ class GRUGradKernel : public framework::OpKernel<T> {
...
-        gru_value.prevOutValue = const_cast<T*>(h0_data);
-        if (h0_grad) {
-          T* h0_grad_data = h0_grad->mutable_data<T>(context.GetPlace());
-          zero(dev_ctx, h0_grad, static_cast<T>(0.0));
-          gru_grad.prevOutGrad = h0_grad_data;
+        if (h0) {
+          gru_value.prevOutValue = ordered_h0.data<T>();
+        } else {
+          gru_value.prevOutValue = nullptr;
+        }
```
Comment (qingqing01) [inline paddle/operators/gru_op.h]:
```cpp
gru_value.prevOutValue = h0 ? ordered_h0.data<T>() : nullptr;
```

File: paddle/operators/gru_op.h [hunk 3]
```diff
@@ -176,11 +194,13 @@ class GRUGradKernel : public framework::OpKernel<T> {
...
+        if (h0 && h0_grad) {
+          gru_grad.prevOutGrad = ordered_h0_grad.data<T>();
```
Comment (qingqing01) [inline paddle/operators/gru_op.h]:
```cpp
gru_grad.prevOutGrad =  h0 && h0_grad ? ordered_h0_grad.data<T>() : nullptr;
```

File: paddle/operators/gru_op.h (hunk 1 above)
Comment (guoshengCS) [reply paddle/operators/gru_op.h]: Done.

File: paddle/operators/gru_op.h (hunk 2 above)
Comment (guoshengCS) [reply paddle/operators/gru_op.h]: Done.

File: paddle/operators/gru_op.h (hunk 3 above)
Comment (guoshengCS) [reply paddle/operators/gru_op.h]: Done.

**Pull request merged**